### DISCUSSION TITLE
How to commit on Github by using VSCode?

### DISCUSSION THREAD

**Discussion by mwasifanwar** [community/community#70114] [New to GitHub]
I forked a repository to add commits to it. I have done all the code work. Kindly do tell me step by step how am I going to commit it to github and open a pull request...

**Comment by SaadARazzaq** [comment]
# Follow the four step process

## 1. Clone the repository
```bash
git clone <repository_url>
```
## 2. Create a new branch
```bash
git checkout -b your-branch-name
```
## 3. Make and commit your changes
```bash
git add .
git commit -m "Your commit message here"
```
## 4. Push your branch to GitHub
```bash
git push origin your-branch-name
```
## Summary:
- Create a Pull Request on GitHub
- Visit your repository on GitHub
- Switch to your branch
- Click "New pull request"
- Set the base and head branches
- Add a title and description
- Review changes in the "Files changed" tab
- Click "Create pull request"
- Describe and submit the pull request

**Reply by mwasifanwar** [reply]
Fantastic You gained a follower

**Reply by SaadARazzaq** [reply]
Thanks man

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
